static-assets/components/cstudio-admin/mods/admin-configurations.js

```diff
@@ -15,476 +15,606 @@
                 }
-              });
-            } else {
-              viewSampleButtonEl.style.display = 'none';
-            }
-
-            CStudioAdminConsole.CommandBar.show();
-
-          } else {
-            editAreaEl.style.display = 'none';
-            CStudioAdminConsole.CommandBar.hide();
+              },
+              requestAsString);
+          } catch (e) {
+            showErrorDialog(e);
           }
-        },
-
-        /*
-        * create editor
-        */
-        setEditor: function (editorContainerEl, readOnly) {
-          var editorEl = document.createElement("pre");
-          editorEl.id = readOnly ? "sample-text" : "text-editor";
-          editorEl.className += "editor-text";
-          editorContainerEl.appendChild(editorEl);
-
-          var mode = "ace/mode/xml";
-
-          var langTools = ace.require("ace/ext/language_tools");
-          var aceEditor = ace.edit(editorEl.id);
-          aceEditor.session.setMode(mode);
-          aceEditor.setOptions({
-            showPrintMargin: false,
-            fontSize: "12px"
-          });
+        } else {
+          CStudioAuthoring.Operations.showSimpleDialog(
+            "errorDialog-dialog",
+            CStudioAuthoring.Operations.simpleDialogTypeINFO,
+            CMgs.format(langBundle, "notification"),
+            CMgs.format(langBundle, "noConfPathError"),
+            null, // use default button
+            YAHOO.widget.SimpleDialog.ICON_BLOCK,
+            "studioDialog"
+          );
+        }
+      }
 
-          if (readOnly) {
-            aceEditor.setReadOnly(true);
-            aceEditor.container.style.background = "#EEEEEE";
+      viewSampleButtonEl.onclick = function () {
+        CStudioAdminConsole.Tool.AdminConfig.prototype.shrinkEditorParent(contentArea, editor);
+        hideSampleButtonEl.style.display = 'inline';
+        viewSampleButtonEl.style.display = 'none';
+        sampleAreaEl.style.display = 'inline';
+      };
+
+      hideSampleButtonEl.onclick = function () {
+        CStudioAdminConsole.Tool.AdminConfig.prototype.expandEditorParent(contentArea, editor);
+        hideSampleButtonEl.style.display = 'none';
+        viewSampleButtonEl.style.display = 'inline';
+        sampleAreaEl.style.display = 'none';
+      };
+
+      $('#encryptButton').click(() => {
+        const editor = this.configInfo.editor;
+        const value = editor.getValue();
+
+        try {
+          const doc = parseValidateDocument(value).documentElement;
+          const tags = doc.querySelectorAll('[encrypted]');
+          const items = findPendingEncryption(tags);
+          if (items.length) {
+            editor.setOption('readOnly', true);
+            editor.container.style.opacity = .5;
+            // Not sure this is supported in all relatively relevant
+            // browsers, test and change or simply change this syntax to es5
+            const { rxjs: { forkJoin, operators: { map } }, util: { ajax } } = CrafterCMSNext;
```

Comment:
Did you test this syntax on firefox and safari at minimum? but preferably edge too.